scripts/install-rekor-cli.sh

```diff
@@ -0,0 +1,151 @@
+    GOBIN=$(go env GOPATH)/bin
+    go install github.com/sigstore/rekor/cmd/rekor-cli@main
+    ln -s "$GOBIN"/rekor-cli "$(eval echo "$INSTALL_DIR")"/rekor-cli
+    exit 0
+  fi
+
+  bootstrap_version='v0.5.0'
+  bootstrap_linux_amd64_sha='a55eba0b526be151ed4b83958d093e925e83c0cda8e15c6508e6d1d360735840'
+  bootstrap_linux_arm_sha='342e0794453cedc72ef92ebaa110b68514c8457c9fcd8fcd64f6b24ed1abba53'
+  bootstrap_linux_arm64_sha='e61850af805f5c12de92e10e90faaac178f1af200b7353f72d03b7fac8e32d8b'
+  bootstrap_darwin_amd64_sha='00621c4ea74347394f7ed5722847259b3dc10256c89a6a568fc1f478f181b83c'
+  bootstrap_darwin_arm64_sha='057def5bf18338fc4d85e889195315d2a815aa5784deb173b13cc41d4150ac15'
+  bootstrap_windows_amd64_sha='af2342f28d9aba6b9c443c9cdc217357d7fd318fac57389e96dd62be0b34005a'
+
+  trap "popd >/dev/null" EXIT
+
+  pushd "$(eval echo "$INSTALL_DIR")" > /dev/null
+
+  case $OS in
+    Linux)
+      case $ARCH in
+        X64)
+          bootstrap_filename='rekor-cli-linux-amd64'
+          bootstrap_sha=${bootstrap_linux_amd64_sha}
+          desired_rekor_cli_filename='rekor-cli-linux-amd64'
+          ;;
+
+        ARM)
+          bootstrap_filename='rekor-cli-linux-arm'
+          bootstrap_sha=${bootstrap_linux_arm_sha}
+          desired_rekor_cli_filename='rekor-cli-linux-arm'
+          ;;
+
+        ARM64)
+          bootstrap_filename='rekor-cli-linux-arm64'
+          bootstrap_sha=${bootstrap_linux_arm64_sha}
+          desired_rekor_cli_filename='rekor-cli-linux-amd64'
+          ;;
+
+        *)
+          log_error "unsupported architecture $ARCH"
+          exit 1
+          ;;
+      esac
+      ;;
+
+    macOS)
+      case $ARCH in
+        X64)
+          bootstrap_filename='rekor-cli-darwin-amd64'
+          bootstrap_sha=${bootstrap_darwin_amd64_sha}
+          desired_rekor_cli_filename='rekor-cli-darwin-amd64'
+          ;;
+
+        ARM64)
+          bootstrap_filename='rekor-cli-darwin-arm64'
+          bootstrap_sha=${bootstrap_darwin_arm64_sha}
+          desired_rekor_cli_filename='rekor-cli-darwin-arm64'
+          ;;
+
+        *)
+          log_error "unsupported architecture $ARCH"
+          exit 1
+          ;;
+      esac
+      ;;
+
+    Windows)
+      case $ARCH in
+        X64)
+          bootstrap_filename='rekor-cli-windows-amd64.exe'
+          bootstrap_sha=${bootstrap_windows_amd64_sha}
+          desired_rekor_cli_filename='rekor-cli-windows-amd64.exe'
+          ;;
+        *)
+          log_error "unsupported architecture $ARCH"
+          exit 1
+          ;;
+      esac
+      ;;
+    *)
+      log_error "unsupported architecture $ARCH"
+      exit 1
+      ;;
+  esac
+
+  expected_bootstrap_version_digest=${bootstrap_sha}
+  log_info "Downloading bootstrap version '${bootstrap_version}' of rekor-cli to verify version to be installed...\n      https://github.com/sigstore/rekor/releases/download/${bootstrap_version}/${bootstrap_filename}"
```

Comment:
I understand that in `install-cosign.sh` a fixed bootstrap cosign version is used to verify the custom cosign version.

Shouldn't `cosign` also be used (instead of `rekor-cli`) to verify the custom `rekor-cli` version? 
I don't think the bootstrap `rekor-cli` binary is ever used and is probably dead code to download it? Maybe I'm missing something here?

Reply:
it is the naming only, i will update the name 😅

Reply:
Thanks for improving the naming!

I still think that this workflow can be simplified, so that we download `rekor-cli` only once. 

I think this file is mostly copy&paste from the `cosign` installation routine, where it makes sense to download some static version of `cosign` for verifiying the desired version.
When running `install-rekor-cli.sh` we already have a verified `cosign` installed. So we can simply download the version of `rekor-cli` we are actually interested in, verify it using `cosign` and exit.

Reply:
makes sense, thanks for bringing this up, i will update that now

Reply:
made the changes @datosh